pandas_queries/q1.py

```diff
@@ -1,50 +1,54 @@
 from datetime import datetime
 
+import pyarrow.compute as pc
+
 from pandas_queries import utils
 
 Q_NUM = 1
 
 
 def q():
     VAR1 = datetime(1998, 9, 2)
+    columns = [
+        "l_quantity",
+        "l_extendedprice",
+        "l_discount",
+        "l_tax",
+        "l_returnflag",
+        "l_linestatus",
+        "l_orderkey",
+    ]
+    kwargs_lineitem = {"filters": pc.field("l_shipdate") <= VAR1}
```

Comment:
This will do the filter already when reading the data. That means that the in-memory query benchmarks are incorrect as they already return a filtered table. This is the case for every query where this occurs.

Reply:
Yeah I know. Any suggestion how you want this organised in general? I can add IF/Else clauses in the code whether the include IO is set or not or add a completely new folder.

Reply:
The idea of the benchmark was to have a single query and that the solution figures out how to do this. With writing manual filters for pyarrow, I am not sure what we benchmark as every tool that supports arrow can do this, but has to duplicate their code with `if else` branches.

Maybe the `INCLUDE_IO` benchmark should only run for tools that do their own predicate and projection pushdowns based on their queries. 

As of now, you basically have a pandas query and a pyarrow + pandas query.

Reply:
You can also say:

```
pd.read_foo()
df = df[list_of_cols]
df = df[filter]
``` 

Is a pattern that should never be used, so what's the point of benchmarking it anyway? This really depends on your PoV. Standalone non I/O benchmarks really don't make much sense for pandas queries since the readers offer all these options that you should use, if you care about performance in any way. Take ``read_sql`` as an example, who would do

```
df = pd.read_sql("select * from something")
df = df[list_of_cols]
df = df[filter]
``` 

instead of structuring a proper SQL query? The functionalities that the arrow reader provides should be viewed in a similar way. Benchmarking "memory operations only" does not make much sense if the I/O time is a significant part of your workflow, which is the case here. This would obviously be different, if you'd have queries that spend 10% of your time reading data and 90% doing transformations.

Reply:
> Is a pattern that should never be used, so what's the point of benchmarking it anyway? 

Sure, but taking that to the extreme. I wouldn't say, I am benchmarking pandas here.

```python
df = pd.read_sql("some long query")  # from snowflake
```

> This would obviously be different, if you'd have queries that spend 10% of your time reading data and 90% doing transformations.

All other tools are able to take a query and run it on in-memory tables and file tables. I think there should be `IF ELSE` cases in the pandas query then.